### ISSUE TITLE
GraphQl. Add a mutation for subscribe feature

### ISSUE THREAD

**Issue by atwixfirster** [magento/magento2#27337]
### Description (*)

There is a subscribe field on a standard Magento storefront:

![27337](https://user-images.githubusercontent.com/13585327/76973533-c0aefd00-6938-11ea-946b-ee17fd924ee6.png)

Currently, GraphQl does not have a mutation which allow to subscribe guest/customer into a newsletter subscription.

### Additional information

A proposed schema:

```
type Mutation {
    subscribeEmailToNewsletter(input: SubscribeInput): SubscribeOutput @resolver(class: "\\Vendor\\ModuleGraphQl\\Model\\Resolver\\SubscribeEmailToNewsletter") @doc(description:"Adds an email into a newsletter subscription")
}

input SubscribeInput {
    email: String! @doc(description: "Email")
}

type SubscribeOutput {
    success: Boolean! @doc(description: "Returns 'true' if an email has been added to newsletter successfully")
    message: String! @doc(description: "Returns a success or an error messages")
}
```

**Example of usage:**

```
mutation {
  subscribeEmailToNewsletter(
    input: {
      email: "user@server.com"
    }
  ) {
    success
    message
  }
}
```

**Response:**
```
{
  "data": {
    "subscribeEmailToNewsletter": {
      "success": true,
      "message": "Thank you for your subscription."
    }
  }
}
```

**Comment by paliarush** [comment]
@atwixfirster, thank you for the proposal.

How about removing wrapper for the input (having `email` as an argument directly)? Such approach will be consistent with https://github.com/magento/architecture/blob/master/design-documents/graph-ql/extensibility.md

Also, why do we need `message` field in the response? How about removing it?

Could I ask you to create a PR with the proposed GraphQL schema in https://github.com/magento/architecture/tree/master/design-documents/graph-ql/coverage ?

Thanks

**Comment by atwixfirster** [comment]
Hi, @paliarush !

First of all thank you for your review.

> How about removing wrapper for the input (having email as an argument directly)? Such approach will be consistent with https://github.com/magento/architecture/blob/master/design-documents/graph-ql/extensibility.md

Ok. I will. 

My proposition has been created by the existing [DevDocs documentation](https://devdocs.magento.com/guides/v2.3/graphql/develop/resolvers.html):

<img width="1455" alt="01" src="https://user-images.githubusercontent.com/13585327/77957701-8e0fe780-72dc-11ea-8feb-b8740e04e11a.png">

> Also, why do we need message field in the response? How about removing it?

The idea of `message` is display message to user. For example,
- [Please enter a valid email address.](https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Newsletter/Controller/Subscriber/NewAction.php#L138)
- [Sorry, but the administrator denied subscription for guests. Please register.](https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Newsletter/Controller/Subscriber/NewAction.php#L121)
- [This email address is already assigned to another user.](https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Newsletter/Controller/Subscriber/NewAction.php#L99)

> Could I ask you to create a PR with the proposed GraphQL schema in https://github.com/magento/architecture/tree/master/design-documents/graph-ql/coverage ?

Can you please provide your clarification for the points above. After that I will create a PR with a correct schema.

Thank you!

**Comment by atwixfirster** [comment]
> Could I ask you to create a PR with the proposed GraphQL schema in https://github.com/magento/architecture/tree/master/design-documents/graph-ql/coverage ?

**done**

https://github.com/magento/architecture/pull/367

**Comment by magento-engcom-team** [comment]
@sdzhepa Thank you for verifying the issue.

Unfortunately, not enough information was provided to acknowledge ticket. Please consider adding the following:
- [ ] Add `"Reproduced on "` label(s) to this ticket based on verification result

Once all required information is added, please add label `"Issue: Confirmed"` again.
Thanks!

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @sdzhepa 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-34288` were created

**Issue Available**: @sdzhepa, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by slavvka** [comment]
Hi @atwixfirster. Thank you for your report. 
The issue has been fixed in magento/magento2#27586 by @atwixfirster in 2.4-develop branch
Related commit(s):
 - [2c812ba13a144f07e4b0716bfa61ef5ccadf76b7](https://github.com/magento/magento2/commit/2c812ba13a144f07e4b0716bfa61ef5ccadf76b7)
 - [670d0b1fcc4e0c49abcbf3bcc0c0ca41e8137cc1](https://github.com/magento/magento2/commit/670d0b1fcc4e0c49abcbf3bcc0c0ca41e8137cc1)
 - [eefac70c0444b85889617453f96d199b3df9608e](https://github.com/magento/magento2/commit/eefac70c0444b85889617453f96d199b3df9608e)
 - [863eb5c2a4d631e57bcb8bc5d177ff9c8a6367c1](https://github.com/magento/magento2/commit/863eb5c2a4d631e57bcb8bc5d177ff9c8a6367c1)
 - [d0d744f70e54d4896cbc3c051c9206591b8f6348](https://github.com/magento/magento2/commit/d0d744f70e54d4896cbc3c051c9206591b8f6348)
 - [59c5d42215171834f8daa349853e7daa06cb8f96](https://github.com/magento/magento2/commit/59c5d42215171834f8daa349853e7daa06cb8f96)
 - [a0a5c2458c518b5b5da2d63548bf704392b0404b](https://github.com/magento/magento2/commit/a0a5c2458c518b5b5da2d63548bf704392b0404b)

The fix will be available with the upcoming 2.4.1 release.

**Comment by slavvka** [comment]
Hi @atwixfirster. Thank you for your report. 
The issue has been fixed in magento/magento2#27586 by @atwixfirster in 2.4-develop branch
Related commit(s):
 - [2c812ba13a144f07e4b0716bfa61ef5ccadf76b7](https://github.com/magento/magento2/commit/2c812ba13a144f07e4b0716bfa61ef5ccadf76b7)
 - [670d0b1fcc4e0c49abcbf3bcc0c0ca41e8137cc1](https://github.com/magento/magento2/commit/670d0b1fcc4e0c49abcbf3bcc0c0ca41e8137cc1)
 - [eefac70c0444b85889617453f96d199b3df9608e](https://github.com/magento/magento2/commit/eefac70c0444b85889617453f96d199b3df9608e)
 - [863eb5c2a4d631e57bcb8bc5d177ff9c8a6367c1](https://github.com/magento/magento2/commit/863eb5c2a4d631e57bcb8bc5d177ff9c8a6367c1)
 - [d0d744f70e54d4896cbc3c051c9206591b8f6348](https://github.com/magento/magento2/commit/d0d744f70e54d4896cbc3c051c9206591b8f6348)
 - [59c5d42215171834f8daa349853e7daa06cb8f96](https://github.com/magento/magento2/commit/59c5d42215171834f8daa349853e7daa06cb8f96)
 - [a0a5c2458c518b5b5da2d63548bf704392b0404b](https://github.com/magento/magento2/commit/a0a5c2458c518b5b5da2d63548bf704392b0404b)

The fix will be available with the upcoming 2.4.1 release.
